src/test/java/seedu/loyaltylift/storage/JsonSerializableAddressBookTest.java

```diff
@@ -50,7 +49,7 @@ public void toModelType_duplicateCustomers_throwsIllegalValueException() throws
     public void toModelType_noSuchCustomer_throwsIllegalValueException() throws Exception {
         JsonSerializableAddressBook dataFromFile = JsonUtil.readJsonFile(CUSTOMER_NOT_FOUND_FILE,
                 JsonSerializableAddressBook.class).get();
-        assertThrows(CustomerNotFoundException.class, dataFromFile::toModelType);
+        assertThrows(IllegalValueException.class, dataFromFile::toModelType);
```

Comment:
I would like to clarify why the change in the exception thrown here. The test validates that a `CustomerNotFound` exception is thrown for when there is a missing customer which the order supposedly reference to. The changes in this PR does not seem to change this behaviour.